src/fontra/views/editor/scene-controller.js

```diff
@@ -1462,3 +1533,67 @@ function splitLocation(location, glyphAxes) {
 
   return { fontLocation, glyphLocation };
 }
+
+export function connectTwoDistinctContours(path, firstPointIndex, secondPointIndex) {
+  let selectedPointIndices = [];
+
+  const [firstContourIndex, firstContourPointIndex] =
+    path.getContourAndPointIndex(firstPointIndex);
+  const [secondContourIndex, secondContourPointIndex] =
+    path.getContourAndPointIndex(secondPointIndex);
+
+  let firstContour = path.getUnpackedContour(firstContourIndex);
+  let secondContour = path.getUnpackedContour(secondContourIndex);
+
+  if (!!firstContourPointIndex == !!secondContourPointIndex) {
+    secondContour.points.reverse();
+  }
+
+  if (!firstContourPointIndex) {
+    [firstContour, secondContour] = [secondContour, firstContour];
+  }
+  let selectedContourPointIndex1 = firstContour.points.length - 1;
+  let selectedContourPointIndex2 = selectedContourPointIndex1 + 1;
+  let loneCubicHandle;
+  const lastPointFirstContour = firstContour.points.at(-1);
+  if (lastPointFirstContour.type) {
+    loneCubicHandle = firstContour.points.pop();
+  }
+
+  const firstPointSecondContour = secondContour.points.at(0);
+  if (firstPointSecondContour.type && !lastPointFirstContour.type) {
+    loneCubicHandle = secondContour.points.shift();
+  }
+
+  if (loneCubicHandle) {
+    const [handle1, handle2] = [lastPointFirstContour, firstPointSecondContour].map(
```

Comment:
This can't be right, because one of these you have just popped/shiffed from the contour. And indeed: I don't see the correct curve appear: one of the handles stays in the same place where I expect it to move.

Reply:
Thanks once again. Should be fixed now.